tests/backup/backup_stork_test.go

```diff
@@ -0,0 +1,573 @@
+		restoreNames                []string
+		periodicSchedulePolicyName  string
+		periodicSchedulePolicyUid   string
+		schPolicyUid                string
+		scheduleName                string
+		scheduleBackupName          string
+		scheduleNames               []string
+	)
+	JustBeforeEach(func() {
+		newAdminNamespace = StorkNamePrefix + "-" + RandomString(5) // Randomly generating the value for custom admin namespace
+		backupName = fmt.Sprintf("%s-%v", BackupNamePrefix, time.Now().Unix())
+		bkpNamespaces = make([]string, 0)
+		restoreName = fmt.Sprintf("%s-%v", RestoreNamePrefix, time.Now().Unix())
+		backupLocationMap = make(map[string]string)
+		labelSelectors = make(map[string]string)
+
+		numDeployments = 5 // 2 apps deployed in 2 namespaces
+		providers = getProviders()
+
+		StartPxBackupTorpedoTest("BackupAndRestoreWithNonExistingAdminNamespace", "Bakcup and restore with non existing custom namespace", nil, 83717, ATrivedi, Q4FY24)
+		log.InfoD(fmt.Sprintf("App list %v", Inst().AppList))
+		scheduledAppContexts = make([]*scheduler.Context, 0)
+		log.InfoD("Starting to deploy applications")
+		for i := 0; i < numDeployments; i++ {
+			log.InfoD(fmt.Sprintf("Iteration %v of deploying applications", i))
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			appContexts := ScheduleApplications(taskName)
+			for _, ctx := range appContexts {
+				ctx.ReadinessTimeout = appReadinessTimeout
+				namespace := GetAppNamespace(ctx, taskName)
+				bkpNamespaces = append(bkpNamespaces, namespace)
+				scheduledAppContexts = append(scheduledAppContexts, ctx)
+			}
+		}
+	})
+	It("Backup and restore with non existing custom admin namespace", func() {
+
+		Step("Validating deployed applications", func() {
+			log.InfoD("Validating deployed applications")
+			ValidateApplications(scheduledAppContexts)
+		})
+		Step("Creating backup location and cloud setting", func() {
+			log.InfoD("Creating backup location and cloud setting")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, provider := range providers {
+				cloudCredName = fmt.Sprintf("%s-%s-%v", "cred", provider, time.Now().Unix())
+				bkpLocationName = fmt.Sprintf("%s-%s-bl", provider, getGlobalBucketName(provider))
+				cloudCredUID = uuid.New()
+				backupLocationUID = uuid.New()
+				backupLocationMap[backupLocationUID] = bkpLocationName
+				err := CreateCloudCredential(provider, cloudCredName, cloudCredUID, orgID, ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of cloud credential named [%s] for org [%s] with [%s] as provider", cloudCredName, orgID, provider))
+				err = CreateBackupLocation(provider, bkpLocationName, backupLocationUID, cloudCredName, cloudCredUID, getGlobalBucketName(provider), orgID, "", true)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating backup location %s", bkpLocationName))
+			}
+		})
+		Step("Creating new admin namespaces", func() {
```

Comment:
Can't we do this in the pipeline itself? Why do we need to do it in testcase?

Reply:
This is specific to this test case, so changing the pipeline might not be required.